gui/system/forms.py

```diff
@@ -1049,51 +1048,26 @@ def __init__(self, *args, **kwargs):
             ['::', '::']
         ] + list(choices.IPChoices(ipv4=False))
 
-    def clean(self):
-        cdata = self.cleaned_data
-
-        # todo: make this and ix-syslogd support udp6
-        if cdata["stg_syslogserver"]:
-            syslogserver = cdata.get("stg_syslogserver")
-            match = re.match("^[\w\.\-]+(\:\d+)?$", syslogserver)
-            if match is None:
-                self._errors['stg_syslogserver'] = self.error_class([_(
-                    "Invalid syslog server format")
-                ])
-
-        proto = cdata.get("stg_guiprotocol")
-        if proto == "http":
-            return cdata
-
-        if not cdata["stg_guicertificate"]:
-            raise forms.ValidationError(
-                "HTTPS specified without certificate")
-        else:
-            certificate_obj = models.Certificate.objects.get(cert_name=cdata["stg_guicertificate"])
-            fingerprint = certificate_obj.get_fingerprint()
-            # using log.error since it logs to /var/log/messages, /var/log/debug.log as well as /dev/console all at once
-            log.error("Fingerprint of the certificate used in the GUI: " + fingerprint)
-        return cdata
+    def middleware_clean(self, update):
+        keys = update.keys()
+        for key in keys:
+            if isinstance(key, str) and key.startswith('gui'):
```

Comment:
In which cases it is not str?

Also can we make it prefix `ui_` with an underscore? I think thats what I suggested in the earlier review, sorry if it wasn't clear.

Reply:
Right now none, but I was thinking in the future just so that we don't have any errors, this condition uses startswith, so check to ensure that the key is a string. I can remove them if you think that would be better?

Reply:
I am sorry, I missed the underscore, I'll have that patched up

Reply:
I dont think its necessary to check for str, key of dicts are always str as far I can tell and I dont envision we ever changing that (there are attribute names)